wizard.py

```diff
@@ -0,0 +1,1038 @@
+    data and inspection data, i.e. a software supply chain (ssc).
+
+  <Returns>
+    Software Supply Chain data, i.e. a dictionary of step and inspection data
+    Note: these are not not actual in-toto Step and Inspection objects
+
+    e.g.:
+    {
+      steps: [
+        {
+          "name": <unique step name>,
+          "cmd": <expected command>,
+          "modifies": <boolean>  # whether the step modifies artifacts or not
+        }, ...
+      ],
+      inspections: [
+        {
+          "name": <unique inspection name>,
+          "cmd": <command to run inspection>,
+          "based_on": <step name whose products are used for that inspection>
+        }
+      ]
+    }
+  """
+  ssc_steps = []
+  ssc_inspections = []
+
+  for step_type in ["vcs", "building", "qa", "package"]:
+    for idx, step in enumerate(session_data.get(step_type, {}).get(
+        "items", [])):
+      # FIXME: Come up with better auto names
+      step_name = "{}-{}".format(step_type, idx + 1)
+      ssc_step = {
+        "name": step_name,
+        "cmd" : step["cmd"],
+        "modifies": True
+      }
+
+      # We create an inspection for each retval, stdout and stderr for
+      # each specified QA step
+      # Special treatment for QA steps:
+      # If the user has specified how s/he knows if a QA step went well, e.g.
+      # by looking at return value, stdout and/or stderr, we use that
+      # information to suggest in-toto inspections (one per posted retval,
+      # stdout, stderr per posted, QA step)
+      if step_type == "qa":
+        # Override modifies property set above
+        # We suppose QA steps don't modify artifacts but just look at them
+        ssc_step["modifies"] = False
+
+        for inspect_type in ["retval", "stdout", "stderr"]:
+          enabled = step.get(inspect_type)
+          val = step.get(inspect_type + "_value")
+          operator = step.get(inspect_type + "_operator")
+
+          # Construct return value or byproducts inspection command
+          # c.f. https://github.com/in-toto/in-toto-inspections
+          if enabled:
+            # Inspection commands require a link file (to inspect)
+            # We use the auto name of the according QA step to
+            # generate the filename passed to the inspection command
+            # FIXME: What about the functionary keyid in the link file name?!
+            link = in_toto.models.link.FILENAME_FORMAT_SHORT.format(
+                step_name=step_name)
+            value = step.get(inspect_type + "_value")
+
+            if inspect_type == "retval":
+              run = ("inspect-return-value --link={link} --{operator} {value}"
+                  .format(link=link, operator=operator, value=value))
+
+            elif inspect_type in ["stdout", "stderr"]:
+              if operator == "empty":
+                operator = "is"
+                value = ""
+
+              run = ("inspect-byproducts"
```

Comment:
What does the code guideline say of using `'` to avoid the `\"`'s?

Reply:
Nothing yet. There is an open [discussion in the code guideline issue tracker](https://github.com/secure-systems-lab/code-style-guidelines/issues/4) though.

Reply:
Ah, ok. Then I guess we can leave it until we decide on it.